iceoryx2-bb/lock-free/tests/spmc_unrestricted_atomic_tests.rs

```diff
@@ -164,3 +169,104 @@ fn spmc_unrestricted_atomic_get_ptr_write_and_update_works_concurrently() {
         pred = v;
     }
 }
+
+#[test]
+fn spmc_unrestricted_atomic_mgmt_release_producer_allows_new_acquire() {
+    let _test_lock = TEST_LOCK.lock().unwrap();
+    let sut = UnrestrictedAtomicMgmt::new();
+
+    let p1 = unsafe { sut.__internal_acquire_producer() };
+    assert_that!(p1, is_ok);
+    assert_that!(unsafe { sut.__internal_acquire_producer() }, is_err);
+
+    unsafe {
+        sut.__internal_release_producer();
+    }
+
+    let p2 = unsafe { sut.__internal_acquire_producer() };
+    assert_that!(p2, is_ok);
+}
+
+#[test]
+fn spmc_unrestricted_atomic_mgmt_get_ptr_write_and_update_works() {
+    let _test_lock = TEST_LOCK.lock().unwrap();
+
+    let value: u64 = 3;
+    let value_ptr: *const u64 = &value;
+    let value_size = core::mem::size_of::<u64>();
+    let value_alignment = core::mem::align_of::<u64>();
+
+    let mut read_value: u64 = 0;
+    let read_value_ptr: *mut u64 = &mut read_value;
+
+    let atomic = UnrestrictedAtomic::<u64>::new(0);
+    let data_ptr = atomic.__internal_get_data_ptr();
+    let mgmt = atomic.__internal_get_mgmt();
+
+    unsafe {
+        assert_that!(mgmt.__internal_acquire_producer(), is_ok);
+        let write_cell_ptr =
+            mgmt.__internal_get_ptr_to_write_cell(value_size, value_alignment, data_ptr);
+        core::ptr::copy_nonoverlapping(value_ptr as *const u8, write_cell_ptr, value_size);
+
+        // new value not read before update
+        mgmt.load(
+            read_value_ptr as *mut u8,
+            value_size,
+            value_alignment,
+            data_ptr,
+        );
+        assert_that!(read_value, eq 0);
+
+        mgmt.__internal_update_write_cell();
+
+        // new value can be read
+        mgmt.load(
+            read_value_ptr as *mut u8,
+            value_size,
+            value_alignment,
+            data_ptr,
+        );
+        assert_that!(read_value, eq value);
+
+        mgmt.__internal_release_producer();
+    }
+}
+
+fn internal_pointer_calculation_works<ValueType: Copy + Default>() {
+    let layout = Layout::new::<UnrestrictedAtomic<ValueType>>();
+
+    unsafe {
+        let random_ptr = alloc(layout) as *mut UnrestrictedAtomic<ValueType>;
+        *(random_ptr) = UnrestrictedAtomic::<ValueType>::new(ValueType::default());
+
+        let mgmt_ptr = addr_of!(*(&*random_ptr).__internal_get_mgmt());
+        let data_ptr = addr_of!(*(&*random_ptr).__internal_get_data_ptr());
+
+        for i in -(align_of::<UnrestrictedAtomic<ValueType>>() as isize) + 1..0 {
```

Comment:
I think the range must be `-(align_of::<UnrestrictedAtomic<ValueType>>() as isize) + 1..=0` otherwise the for loop will check nothing if the alignment is 1 and a correctly aligned pointer shall be also handled correctly.